File: docsapp/com.ibm.concord.platform/src/com/ibm/concord/platform/util/HtmlTag.java
/* ***************************************************************** */
/*                                                                   */
/* IBM Confidential                                                  */
/*                                                                   */
/* IBM Docs Source Materials                                         */
/*                                                                   */
/* (c) Copyright IBM Corporation 2012. All Rights Reserved.          */
/*                                                                   */
/* ***************************************************************** */

package com.ibm.concord.platform.util;

import java.util.ArrayList;
import java.util.List;

/**
 * @author gaowwei@cn.ibm.com
 *
 */
public class HtmlTag
{
  private String tagName;
  private List<Attribute> attributes;
  
  public static class Attribute {
    public String name;
    public String value;
    public Attribute(String n, String v)
    {
      name = n;
      value = v;
    }
  }
  HtmlTag()
  {
    tagName = "";
    attributes = new ArrayList<Attribute>();
  }
  
  public String getTagName()
  {
    return tagName;
  }
  
  public void setTagName(String tagName)
  {
    this.tagName = tagName;
  }
  
  public List<Attribute> getAttributes()
  {
    return attributes;
  }
  
  public void addAttribute(String n, String v)
  {
    Attribute attr = new Attribute(n, v);
    attributes.add(attr);
  }
}

File: docs-web.resources/com.ibm.docs.web.resources/WebContent/js/concord/util/HtmlTag.js
/* ***************************************************************** */
/*                                                                   */
/* IBM Confidential                                                  */
/*                                                                   */
/* IBM Docs Source Materials                                         */
/*                                                                   */
/* (c) Copyright IBM Corporation 2012. All Rights Reserved.          */
/*                                                                   */
/* U.S. Government Users Restricted Rights: Use, duplication or      */
/* disclosure restricted by GSA ADP Schedule Contract with IBM Corp. */
/*                                                                   */
/* ***************************************************************** */

dojo.provide("concord.util.HtmlTag");
dojo.provide("concord.util.HtmlTag.Attribute");

dojo.declare("concord.util.HtmlTag", null, {
	tagName: null,
	attributes: null,

	constructor: function()
	{
		this.tagName = "";
		this.attributes = new Array();
	},

	getTagName: function()
	{
		return this.tagName;
	},

	setTagName: function(t)
	{
		this.tagName = t;
	},

	getAttributes: function()
	{
		return this.attributes;
	},

	addAttribute: function(n, v)
	{
		var attr = new concord.util.HtmlTag.Attribute(n, v);
		this.attributes.push(attr);
	}
});

dojo.declare("concord.util.HtmlTag.Attribute", null, {
	name: null,
	value: null,
	constructor: function(n, v)
	{
		this.name = n;
		this.value = v;
	}
});
